### PR TITLE
[staging] openblas: 0.3.15 -> 0.3.17

### PR DESCRIPTION
**Pull request by r-burns**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by markuskowa** [comment]
Changelog: https://github.com/xianyi/OpenBLAS/releases

File: pkgs/development/libraries/science/math/openblas/default.nix [hunk 1]
```diff
@@ -140,20 +140,6 @@ stdenv.mkDerivation rec {
     sha256 = "1qjr02cqncv20abdp1yzr55n7smhx6h9chqvb0xbp18byynvj87w";
```
Comment (ttuegel) [inline pkgs/development/libraries/science/math/openblas/default.nix:140]: You need to update the `sha256`, or it will just build the old version.

**Review by ttuegel — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/science/math/openblas/default.nix (hunk 1 above)
Comment (r-burns) [reply pkgs/development/libraries/science/math/openblas/default.nix]: Thank you :facepalm:

**Comment by risicle** [comment]
A sneaky cherry-pick back to master builds `openblas` `openblasCompat` `python39Packages.numpy` `python39Packages.scipy` successfully on macos 10.15.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133694` at 6bc10e34 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>45 packages marked as broken and skipped:</summary>
  <ul>
    <li>blastem</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>curaByDagoma</li>
    <li>cura_stable</li>
    <li>displaycal</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>gnuradio3_7Packages.rds</li>
    <li>googleearth-pro</li>
    <li>grass</li>
    <li>hadoop</li>
    <li>hadoop_2_7</li>
    <li>hadoop_2_8</li>
    <li>hadoop_2_9</li>
    <li>hplipWithPlugin_3_16_11</li>
    <li>hplip_3_16_11</li>
    <li>impressive</li>
    <li>invoice2data</li>
    <li>julia</li>
    <li>julia-lts</li>
    <li>julia_10</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>loxodo</li>
    <li>metamorphose2</li>
    <li>ocropus</li>
    <li>odpdown</li>
    <li>opentsdb</li>
    <li>pig</li>
    <li>plover.stable</li>
    <li>rainbowstream</li>
    <li>scribus</li>
    <li>scyther</li>
    <li>spark</li>
    <li>termplay</li>
    <li>torchat</li>
    <li>unifi-video</li>
    <li>volatility</li>
    <li>winpdb</li>
  </ul>
</details>
<details>
  <summary>3912 packages skipped due to time constraints:</summary>
  <ul>
    <li>DisnixWebService</li>
    <li>MACS2</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>abcl</li>
    <li>actdiag (python39Packages.actdiag)</li>
    <li>adafruit-ampy</li>
    <li>adapta-gtk-theme</li>
    <li>adoptopenjdk-icedtea-web</li>
    <li>aiodnsbrute</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>openblas</li>
    <li>openblasCompat</li>
  </ul>
</details>

**Comment by risicle** [comment]
@ofborg build openblas openblasCompat python39Packages.numpy python39Packages.scipy

**Review by markuskowa — APPROVED** [review]

**Pull request merged**